### PR TITLE
add combined local/remote cache support

### PR DESCRIPTION
**Pull request by emmanuelm41**

### CODE REVIEW

**Review by lucaslopezf — COMMENTED** [review]
Just a few comments

File: pkg/zcache/combined_cache.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+func (c *combinedCache) Set(ctx context.Context, key string, value interface{}, remoteCacheTtl time.Duration) error {
+	if err := c.remoteCache.Set(ctx, key, value, remoteCacheTtl); err != nil && !c.isRemoteBestEffort {
+		return err
+	}
+
+	// ttl is controlled by cache instantiation, so it does not matter here
+	if err := c.localCache.Set(ctx, key, value, -1); err != nil {
+		return err
```
Comment (lucaslopezf) [inline pkg/zcache/combined_cache.go:25]: We should add metrics know if synchronization is lost between the cache, what do you think?

File: pkg/zcache/combined_cache.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	ZCache
+}
+
+type combinedCache struct {
+	localCache         LocalCache
+	remoteCache        RemoteCache
+	isRemoteBestEffort bool
+}
+
+func (c *combinedCache) Set(ctx context.Context, key string, value interface{}, remoteCacheTtl time.Duration) error {
```
[the remark this answers is not in the document]
Comment (emmanuelm41) [reply pkg/zcache/combined_cache.go:18]: Check it now!

File: pkg/zcache/combined_cache.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	}
+
+	return nil
+}
+
+func (c *combinedCache) Delete(ctx context.Context, key string) error {
+	err1 := c.localCache.Delete(ctx, key)
+	err2 := c.remoteCache.Delete(ctx, key)
+
+	if err1 != nil {
```
[the remark this answers is not in the document]
Comment (emmanuelm41) [reply pkg/zcache/combined_cache.go:51]: What do you think now?

File: pkg/zcache/combined_cache.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+		return err1
+	}
+
+	return nil
+}
+
+func (c *combinedCache) GetStats() ZCacheStats {
+	localStats := c.localCache.(interface{}).(*bigcache.BigCache).Stats()
+	remoteStats := c.remoteCache.(interface{}).(*redis.Client).PoolStats()
+	return ZCacheStats{Bigcache: &localStats, Redis: remoteStats}
```
Comment (lucaslopezf) [inline pkg/zcache/combined_cache.go]: linter

**Review by lucaslopezf — COMMENTED** [review]
Great! Only two comments

File: pkg/zcache/redis_cache.go [hunk 5]
```diff
@@ -85,3 +93,33 @@ func (c *redisCache) HSet(ctx context.Context, key string, values ...interface{}
 func (c *redisCache) HGet(ctx context.Context, key, field string) (string, error) {
 	return c.client.HGet(ctx, key, field).Result()
 }
+
+func (c *redisCache) GetStats() ZCacheStats {
+	poolStats := c.client.PoolStats()
+	// ctx := context.Background()
```
Comment (lucaslopezf) [inline pkg/zcache/redis_cache.go:99]: Are these comments useful?

Comment (emmanuelm41) [reply pkg/zcache/redis_cache.go:99]: Not really... we can take statistics from Redis directly with INFO command. After that, we can parse it and take what we want. I left them there in order to complete it later

**Review by lucaslopezf — APPROVED** [review]

**Pull request merged**